[GitHub Desktop] Add new working with a fork flow (#3210)
github/docs: content/desktop/contributing-and-collaborating-using-github-desktop/cloning-and-forking-repositories-from-github-desktop.md

  free-pro-team: '*'
---

### About local repositories

> Repositories on {% data variables.product.prodname_dotcom %} are remote repositories that you can clone or fork with {% data variables.product.prodname_desktop %} to create a local repository.

Reviewer: Again, I know wrote this in a suggested change, but my writing process typically involves rewriting most single sentences a few times. 😂  On that note... you can always tweak suggested changes before (or after) applying them. I personally have to see everything together on staging to know if it works, which I can't do when leaving suggested changes. So I have no expectation that my changes will be applied verbatim!

Also, if you can tell before applying that the suggestion needs to be tweaked (such as the typo below) you can actually edit the comment to update the change before applying it!

```suggestion
Repositories on {% data variables.product.prodname_dotcom %} are remote repositories. You can clone or fork a repository with {% data variables.product.prodname_desktop %} to create a local repository on your computer.
```

> You can create a local copy of any repository on {% data variables.product.product_name %} that you have access to by cloning the repository. If you own a repository or are a have write permissions, you can sync between the local and remote locations. For more information, see "[Syncing your branch](/desktop/contributing-and-collaborating-using-github-desktop/syncing-your-branch)."

Reviewer: I assume this was my own typo in a previous suggested change. 😆 

```suggestion
You can create a local copy of any repository on {% data variables.product.product_name %} that you have access to by cloning the repository. If you own a repository or have write permissions, you can sync between the local and remote locations. For more information, see "[Syncing your branch](/desktop/contributing-and-collaborating-using-github-desktop/syncing-your-branch)."
```

> When you clone a repository, any changes you push to {% data variables.product.product_name %} will affect the original repository. To make changes without affecting the original project, you can create a separate copy by forking the repository. You can open a pull request to propose that maintainers incorporate the changes in your fork to the original upstream repository. For more information, see "[About forks](/github/collaborating-with-issues-and-pull-requests/about-forks)."

Reviewer: ```suggestion
When you clone a repository, any changes you push to {% data variables.product.product_name %} will affect the original repository. To make changes without affecting the original project, you can create a separate copy by forking the repository. You can create a pull request to propose that maintainers incorporate the changes in your fork into the original upstream repository. For more information, see "[About forks](/github/collaborating-with-issues-and-pull-requests/about-forks)."
```

{% endwindows %}

### Forking repositories
{% mac %}

{% data reusables.desktop.fork-type-prompt %}

>   ![Menu to select if a fork will contribute to an upstream repository](/assets/images/help/desktop/mac-fork-options-prompt.png)

Reviewer: Is there a reason to have the screenshots outside of the reusables? I haven't worked on Desktop in a long time so I'm not sure if this would actually work, but could you use the `{% mac %}` and `{% endmac %}` tags within each reusables to version the screenshots by OS, and then only have one procedure here?

{% data reusables.desktop.open-repository-settings %}
  ![Repository settings menu option in the Mac app](/assets/images/help/desktop/mac-repository-settings.png)
{% data reusables.desktop.select-fork-behavior %}
  ![Options in the fork behavior menu](/assets/images/help/desktop/mac-fork-behavior-menu.png)

{% endmac %}

{% windows %}

{% data reusables.desktop.fork-type-prompt %}
  ![Menu to select if a fork will contribute to an upstream repository](/assets/images/help/desktop/windows-fork-options-prompt.png)

> {% data reusables.desktop.open-repository-settings %}

Reviewer: This and the below should be a separate procedure... it's not part of creating a fork. Let's add a new header called something like "Managing fork behavior" and link to it in the conceptual section as mentioned in my earlier comment.
